gto/cli.py

```diff
@@ -562,6 +566,9 @@ def assign(
     message: Optional[str] = option_message,
     simple: str = option_simple,
     force: bool = option_force,
+    auto_push: bool = Option(
```

Comment:
Looks like `assign --auto-push` doesn't push the first Git tag that is created to register a version:
```bash
$ gto promote ohmypromotion --stage dev --auto-push
Created git tag 'ohmypromotion@v0.0.1' that registers version
To push the changes upstream, run:
    git push origin ohmypromotion@v0.0.1
Created git tag 'ohmypromotion#dev#1' that assigns stage to version 'v0.0.1'
Running `git push origin ohmypromotion#dev#1`
Successfully pushed git tag ohmypromotion#dev#1 to the remote.
```

Reply:
Open question: is it better to do it like this (create tag, push tag, create tag, push tag), or do it together (create both tags, push both)?

Reply:
For some reason it doesn't fail if `--stage` arg is not provided, causing this:
```shell
$ gto assign bugfixed --auto-push
Created git tag 'bugfixed@v0.0.1' that registers version
Running `git push origin bugfixed@v0.0.1`
Successfully pushed git tag bugfixed@v0.0.1 to the remote.
Created git tag 'bugfixed#None#1' that assigns stage to version 'v0.0.1'
Running `git push origin bugfixed#None#1`
Successfully pushed git tag bugfixed#None#1 to the remote.
❌ Cannot parse tag name 'bugfixed#None#1'
```

Reply:
mmm... I can't see how it can be related to what I have done in this PR. Behavior is same without the `--auto-push`, can you confirm? This is something that probably should be fixed in a separate PR?

Perhaps now is more urgent because one may push invalid tags (like `bugfixed#None#1`)?

Reply:
Yes, it doesn't depend on `--auto-push`. We can do it in a separate PR for sure. Created a separate issue for that https://github.com/iterative/gto/issues/285